tests/test_client.py

```diff
@@ -33,79 +34,186 @@ def test_eval_calls_request(self, MockPath):
+                client.advanced_search(
+                    q="my-query",
+                    court="ewhc",
+                    judge="a. judge",
+                    party="a party",
+                    page=2,
+                    page_size=20,
+                    show_unpublished=True
+                )
+
+                expected_vars = {
+                    "court": "ewhc",
+                    "judge": "a. judge",
+                    "page": 2,
+                    "page-size": 20,
+                    "q": "my-query",
+                    "party": "a party",
+                    "neutral_citation": "",
+                    "specific_keyword": "",
+                    "order": "",
+                    "from": "",
+                    "to": "",
+                    "show_unpublished": "true",
+                    "only_unpublished": "false"
+                }
+
+                client.invoke.assert_called_with(
+                    '/judgments/search/search.xqy',
+                    json.dumps(expected_vars)
+                )
+
+    def test_advanced_search_user_cannot_view_unpublished_but_show_unpublished_is_true(self):
+        """The user is not permitted to see unpublished judgments but is attempting to view them
+        Set `show_unpublished` to false and log a warning"""
+        client = MarklogicApiClient("", "", "", False)
 
-    def test_eval_xslt_with_default(self):
+        with patch.object(client, 'invoke'):
+            with patch.object(client, 'user_can_view_unpublished_judgments', return_value=False):
+                with patch.object(logging, "warning") as mock_logging:
+                    client.advanced_search(
+                        q="my-query",
+                        court="ewhc",
+                        judge="a. judge",
+                        party="a party",
+                        page=2,
+                        page_size=20,
+                        show_unpublished=True
+                    )
+
+                    expected_vars = {
+                        "court": "ewhc",
+                        "judge": "a. judge",
+                        "page": 2,
+                        "page-size": 20,
+                        "q": "my-query",
+                        "party": "a party",
+                        "neutral_citation": "",
+                        "specific_keyword": "",
+                        "order": "",
+                        "from": "",
+                        "to": "",
+                        "show_unpublished": "false",
+                        "only_unpublished": "false"
+                    }
+
+                    client.invoke.assert_called_with(
+                        '/judgments/search/search.xqy',
+                        json.dumps(expected_vars)
+                    )
+                    mock_logging.assert_called()
+
+    def test_eval_xslt_with_default_and_user_can_view_unpublished(self):
```

Comment:
I'm not sure what 'default' is in this context

Reply:
Oh, it's "show_unpublished" being true, in the meaning of "please show unpublished (if I'm allowed)"

Reply:
No, "default" is the default XSLT as opposed to the accessible one 🤦  I realise these are so confusing. I will look into contexts/better naming.